classes/article/ArticleGalleyDAO.inc.php

```diff
@@ -235,27 +248,49 @@ function _fromRow($row) {
 	function insertObject($galley) {
 		$this->update(
 			'INSERT INTO submission_galleys
-				(submission_id, label, locale, seq, remote_url, file_id)
+				(submission_id, label, locale, seq, remote_url)
 				VALUES
-				(?, ?, ?, ?, ?, ?)',
+				(?, ?, ?, ?, ?)',
 			array(
 				(int) $galley->getSubmissionId(),
 				$galley->getLabel(),
 				$galley->getLocale(),
 				$galley->getSequence() == null ? $this->getNextGalleySequence($galley->getSubmissionId()) : $galley->getSequence(),
 				$galley->getRemoteURL(),
-				$galley->getFileId(),
 			)
 		);
 		$galley->setId($this->getInsertId());
 		$this->updateLocaleFields($galley);
+		$this->addFile($galley->getId(), $galley->getFileId(), $galley->getCurrentVersionId());
 
 		HookRegistry::call('ArticleGalleyDAO::insertNewGalley', array(&$galley, $galley->getId()));
 
 		return $galley->getId();
 	}
 
 	/**
+	* Connect galley version with file in submission_galley_files
+	* @param $galleyId ArticleGalleyId
+	* @param $fileId FileId
+	* @param $version SubmissionRevision
+	*/
+	function addFile($galleyId, $fileId=0, $version=1){
+		$this->update(
+			'INSERT INTO submission_galley_files
+				(galley_id, file_id, version)
+				VALUES
+				(?, ?, ?)
+			ON DUPLICATE KEY UPDATE
```

Comment:
It seems that not all DB engines act in the same way on this command, e.g. see MySql documentation: "The effects are not identical for an InnoDB table where a is an auto-increment column. With an auto-increment column, an INSERT statement increases the auto-increment value but UPDATE does not.". Also, PostgreSQL seem to have different command. Thus it wold be better to differentiate it into insert and update, I believe...